Misc/NEWS.d/next/Library/2023-12-17-10-22-55.gh-issue-112182.jLWGlr.rst

```diff
@@ -0,0 +1,2 @@
+:meth:`asyncio.futures.Future.set_exception()` now silently transforms
+``StopIteration`` into ``RuntimeError``. Patch contributed by Jamie Phan.
```

Comment:
Maybe add "instead of hanging or other misbehavior". You might even drop "silently": "silently" doing things is often considered a code smell in Python, see the Zen of Python -- but here it is a positive property.